### PR TITLE
Refinement strategy

### PR DESCRIPTION
**Pull request by zbmorrow**
This pull request adds the functions and data structures that will be used in the refinement strategy. All functions are currently empty.

### CODE REVIEW

File: SparseGrids/tsgGridFourier.hpp [hunk 1]
```diff
@@ -183,6 +189,11 @@ class GridFourier : public BaseCanonicalGrid {
     MultiIndexSet tensors;
     MultiIndexSet active_tensors;
     std::vector<int> active_w;
+
+    MultiIndexSet updated_tensors;
```
Comment (mkstoyanov) [inline SparseGrids/tsgGridFourier.hpp]: Leave the variables out of this
* having unused variables will generate an unnecessary warning
* the variables should be added to the file I/O, (make another PR with just variables and file I/O)

Comment (zbmorrow) [reply SparseGrids/tsgGridFourier.hpp]: Done
